src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Match.cs

```diff
@@ -78,10 +78,8 @@ internal void Reset(Regex regex, string text, int textbeg, int textend, int text
             _textstart = textstart;
 
             int[] matchcount = _matchcount;
-            for (int i = 0; i < matchcount.Length; i++)
-            {
-                matchcount[i] = 0;
-            }
+            _ = matchcount.Length; // dereference to avoid null check on below line
+            matchcount.AsSpan().Clear();
```

Comment:
What is the typical `matchcount.Length`? Is this going to be an improvement in typical case?